docs/api-reference/webgl/framebuffer.md

```diff
@@ -266,6 +268,23 @@ This function makes calls to the following WebGL APIs:
 
 [gl.readPixels](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/readPixels), [`gl.bindFramebuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindFramebuffer)
 
+### readPixelsAsync (WebGL2)
```

Comment:
I really like this method.

Q: Do we want to call it async? 

* It is async in a sense, but not the usual javascript sense.
* For one thing, there is no promise or callback returned. 
* There is no way for JS to know when it completes, except that it will always be completed before next operation. The GPU will sync it whenever used.  It will be synced when reading the data from the buffer. etc.

Also note that Chrome is adding a true [getBufferDataAsync extension](https://github.com/KhronosGroup/WebGL/blob/master/sdk/tests/conformance2/extensions/webgl-get-buffer-sub-data-async.html)

Something like
```js
promise = WEBGL_get_buffer_sub_data_async.getBufferSubDataAsync(gl.ARRAY_BUFFER, 0, new ArrayBuffer(4))
```